include/revng/StackAnalysis/ABI/CallingConventionTrait.h

```diff
@@ -0,0 +1,645 @@
+#pragma once
+
+//
+// This file is distributed under the MIT License. See LICENSE.md for details.
+//
+
+#include "revng/Model/Binary.h"
+#include "revng/Model/Register.h"
+#include "revng/Model/Type.h"
+
+namespace abi {
+
+/// A trait struct which is specialized for all of the supported ABIs.
+/// It describes the specifics of the calling convention for the ABI.
+template<model::abi::Values V>
+struct CallingConventionTrait {
+
+  /// States whether specified ABI allows using both generic and vector
+  /// registers in parallel.
```

Comment:
Uhm, not sure I understand. Is this for the fact that the argument index of GPR (General Purpose Registers) evolve independently w.r.t. the vector/floating point ones?

It's not very clear.